bootstrap/data_ingester.py

```diff
@@ -25,53 +25,49 @@
 CONTENT_TYPE = "application/json"
 
 
-def get_dataset_id(create_dataset_url, headers, dataset_title, schema_id):
+def get_dataset_id(create_dataset_url, headers, dataset_title, schema_id, data):
     """
     Get the datasetId by making a POST Request to "/data/foundation/catalog/datasets?requestDataSource=true"
     :param create_dataset_url: url
     :param headers: headers
     :param dataset_title: dataset title
     :param schema_id: schema url
+    :param data: post request data
     :return: dataset id
     """
-    data_for_create_dataset = {
-        "schemaRef": {
-            "id": schema_id,
-            "contentType": "application/vnd.adobe.xed+json; version=1"
-        },
-        "name": dataset_title,
-        "description": dataset_title,
-        "fileDescription": {
-            "persisted": True,
-            "containerFormat": "parquet",
-            "format": "json"
-        },
-        "aspect": "production"
-    }
+
+    # Set the title and description
+    data['name'] = dataset_title
+    data['description'] = dataset_title
+
+    # Set the schema id
+    data["schemaRef"]["id"] = schema_id
+
     headers["Accept"] = CONTENT_TYPE
     headers["Content-Type"] = CONTENT_TYPE
-    res_text = http_request("post", create_dataset_url, headers, json.dumps(data_for_create_dataset))
+    res_text = http_request("post", create_dataset_url, headers, json.dumps(data))
     dataset_response = str(json.loads(res_text))
     LOGGER.debug("dataset_response is %s", dataset_response)
     dataset_id = dataset_response.split("@/dataSets/")[1].split("'")[0]
     LOGGER.debug("dataset_id = %s", dataset_id)
     return dataset_id
 
 
-def get_batch_id(create_batch_url, headers, dataset_id):
+def get_batch_id(create_batch_url, headers, dataset_id, data):
 
     """
     Get the batchId by making a POST request to "/data/foundation/import/batches"
     :param create_batch_url: url
     :param headers: headers
     :param dataset_id: dataset id
+    :param data: post request data
     :return: batch id
     """
-    data_for_create_batch = {"datasetId": dataset_id}
+    data["datasetId"] = dataset_id
```

Comment:
Would it be normal to check for null/empty `data` param before de-referencing it?